### PR TITLE
All code snippets for v1 messages

### PR DESCRIPTION
**Pull request by SecondeJK**
This adds all the code snippets for Messages v1

### CODE REVIEW

File: messages/messenger/send-audio.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+$client = new \Vonage\Client($keypair);
+
+$audioObject = new \Vonage\Messages\MessageObjects\AudioObject(
+    'https://example.com/audio.mp3',
+    'This is an audio file'
+);
+
+$sms = new \Vonage\Messages\MessageType\Messenger\MessengerAudio(
```
Comment (superchilled) [inline messages/messenger/send-audio.php:17]:
```suggestion
$message = new \Vonage\Messages\MessageType\Messenger\MessengerAudio(
```

Or some other variable name. `$sms` doesn't seem appropriate for this snippet.

File: messages/messenger/send-file.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    VONAGE_APPLICATION_ID
+);
+
+$client = new \Vonage\Client($keypair);
+
+$fileObject = new \Vonage\Messages\MessageObjects\FileObject(
+    'https://example.com/file.pdf',
+);
+
+$sms = new \Vonage\Messages\MessageType\Messenger\MessengerFile(
```
Comment (superchilled) [inline messages/messenger/send-file.php:16]:
```suggestion
$message = new \Vonage\Messages\MessageType\Messenger\MessengerFile(
```

Ditto

File: messages/messenger/send-image.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+$client = new \Vonage\Client($keypair);
+
+$imageObject = new \Vonage\Messages\MessageObjects\ImageObject(
+    'https://example.com/image.jpg',
+    'This is an image'
+);
+
+$sms = new \Vonage\Messages\MessageType\Messenger\MessengerImage(
```
Comment (superchilled) [inline messages/messenger/send-image.php:17]:
```suggestion
$message = new \Vonage\Messages\MessageType\Messenger\MessengerImage(
```

File: messages/messenger/send-text.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+require_once __DIR__ . '../../vendor/autoload.php';
+
+$keypair = new \Vonage\Client\Credentials\Keypair(
+    file_get_contents(VONAGE_APPLICATION_PRIVATE_KEY_PATH),
+    VONAGE_APPLICATION_ID
+);
+
+$client = new \Vonage\Client($keypair);
+
+$sms = new \Vonage\Messages\MessageType\Messenger\MessengerText(
```
Comment (superchilled) [inline messages/messenger/send-text.php:12]:
```suggestion
$message = new \Vonage\Messages\MessageType\Messenger\MessengerText(
```

File: messages/messenger/send-video.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+$client = new \Vonage\Client($keypair);
+
+$videoObject = new \Vonage\Messages\MessageObjects\VideoObject(
+    'https://example.com/video.mp4',
+    'This is an video file'
+);
+
+$sms = new \Vonage\Messages\MessageType\Messenger\MessengerVideo(
```
Comment (superchilled) [inline messages/messenger/send-video.php:17]:
```suggestion
$message = new \Vonage\Messages\MessageType\Messenger\MessengerVideo(
```

File: messages/mms/send-mms.php [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+$client = new \Vonage\Client($keypair);
+
+$image = new \Vonage\Messages\MessageObjects\ImageObject(
+    'https://example.com/image.jpg',
+    'A MMS image message, with caption, sent using the Vonage Messages API'
+);
+
+$sms = new \Vonage\Messages\MessageType\MMS\MMSImage(
```
Comment (superchilled) [inline messages/mms/send-mms.php:17]:
```suggestion
$mms = new \Vonage\Messages\MessageType\MMS\MMSImage(
```

File: messages/viber/send-image.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+$client = new \Vonage\Client($keypair);
+
+$imageObject = new \Vonage\Messages\MessageObjects\ImageObject(
+    'https://example.com/image.jpg',
+    'This is an image'
+);
+
+$sms = new \Vonage\Messages\MessageType\Viber\ViberImage(
```
Comment (superchilled) [inline messages/viber/send-image.php:17]:
```suggestion
$viber = new \Vonage\Messages\MessageType\Viber\ViberImage(
```

File: messages/viber/send-text.php [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+require_once __DIR__ . '../../vendor/autoload.php';
+
+$keypair = new \Vonage\Client\Credentials\Keypair(
+    file_get_contents(VONAGE_APPLICATION_PRIVATE_KEY_PATH),
+    VONAGE_APPLICATION_ID
+);
+
+$client = new \Vonage\Client($keypair);
+
+$sms = new \Vonage\Messages\MessageType\Viber\ViberText(
```
Comment (superchilled) [inline messages/viber/send-text.php:12]:
```suggestion
$viber = new \Vonage\Messages\MessageType\Viber\ViberText(
```

File: messages/whatsapp/send-audio.php [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+$client = new \Vonage\Client($keypair);
+
+$audioObject = new \Vonage\Messages\MessageObjects\AudioObject(
+    'https://example.com/audio.mp3',
+    'This is an audio file'
+);
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppAudio(
```
Comment (superchilled) [inline messages/whatsapp/send-audio.php:17]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppAudio(
```

File: messages/whatsapp/send-button-link.php [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+                "sub_type" => "url",
+                "parameters" => [
+                    ["type" => "text", "text" => "1Z999AA10123456784"],
+                ],
+            ],
+        ],
+    ],
+];
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```
Comment (superchilled) [inline messages/whatsapp/send-button-link.php:49]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```

File: messages/whatsapp/send-button-quick-reply.php [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+                "index" => 1,
+                "parameters" => [
+                    ["type" => "payload", "payload" => "No-Button-Payload"],
+                ],
+            ],
+        ],
+    ],
+];
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```
Comment (superchilled) [inline messages/whatsapp/send-button-quick-reply.php:54]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```

File: messages/whatsapp/send-contact.php [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+                    "type" => "WORK",
+                    "wa_id" => "16505551234",
+                ],
+            ],
+            "urls" => [["url" => "https://www.facebook.com", "type" => "WORK"]],
+        ],
+    ],
+];
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```
Comment (superchilled) [inline messages/whatsapp/send-contact.php:64]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```

File: messages/whatsapp/send-file.php [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+$client = new \Vonage\Client($keypair);
+
+$fileObject = new \Vonage\Messages\MessageObjects\FileObject(
+    'https://example.com/file.pdf',
+    'This is a whatsapp file'
+);
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppFile(
```
Comment (superchilled) [inline messages/whatsapp/send-file.php:17]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppFile(
```

File: messages/whatsapp/send-image.php [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+$client = new \Vonage\Client($keypair);
+
+$imageObject = new \Vonage\Messages\MessageObjects\ImageObject(
+    'https://example.com/image.jpg',
+    'This is a whatsapp image'
+);
+
+$sms = new \Vonage\Messages\MessageType\Messenger\MessengerImage(
```
Comment (superchilled) [inline messages/whatsapp/send-image.php:17]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\Messenger\MessengerImage(
```

File: messages/whatsapp/send-location.php [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+    "type" => "location",
+    "location" => [
+        "longitude" => -122.425332,
+        "latitude" => 37.758056,
+        "name" => "Facebook HQ",
+        "address" => "1 Hacker Way, Menlo Park, CA 94025",
+    ],
+];
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```
Comment (superchilled) [inline messages/whatsapp/send-location.php:22]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```

File: messages/whatsapp/send-media-mtm.php [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+                    ["type" => "text", "text" => "Value 1"],
+                    ["type" => "text", "text" => "Value 2"],
+                    ["type" => "text", "text" => "Value 3"],
+                ],
+            ],
+        ],
+    ],
+];
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```
Comment (superchilled) [inline messages/whatsapp/send-media-mtm.php:45]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```

File: messages/whatsapp/send-mtm.php [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+$custom = [
+    "name" => WHATSAPP_TEMPLATE_NAMESPACE . ":" . WHATSAPP_TEMPLATE_NAME,
+    "parameters" => [
+        ["default" => "Vonage Verification"],
+        ["default" => "64873"],
+        ["default" => "10"],
+    ],
+];
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```
Comment (superchilled) [inline messages/whatsapp/send-mtm.php:21]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppCustom(
```

File: messages/whatsapp/send-text.php [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+require_once __DIR__ . '../../vendor/autoload.php';
+
+$keypair = new \Vonage\Client\Credentials\Keypair(
+    file_get_contents(VONAGE_APPLICATION_PRIVATE_KEY_PATH),
+    VONAGE_APPLICATION_ID
+);
+
+$client = new \Vonage\Client($keypair);
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppText(
```
Comment (superchilled) [inline messages/whatsapp/send-text.php:12]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppText(
```

File: messages/whatsapp/send-video.php [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+$client = new \Vonage\Client($keypair);
+
+$videoObject = new \Vonage\Messages\MessageObjects\VideoObject(
+    'https://example.com/video.mp4',
+    'This is an video file'
+);
+
+$sms = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppVideo(
```
Comment (superchilled) [inline messages/whatsapp/send-video.php:17]:
```suggestion
$whatsApp = new \Vonage\Messages\MessageType\WhatsApp\WhatsAppVideo(
```

File: messages/whatsapp/send-button-link.php [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+$client = new \Vonage\Client($keypair);
+
+$custom = [
+    "type" => "template",
+    "template" => [
+        "namespace" => WHATSAPP_TEMPLATE_NAMESPACE,
+        "name" => WHATSAPP_TEMPLATE_NAME,
+        "language" => ["code" => "en", "policy" => "deterministic"],
+        "components" => [
+            [
```
Comment (superchilled) [inline messages/whatsapp/send-button-link.php:19]: Do the inner arrays here get converted to JSON objects? Note: the JSON for components uses both arrays _and_ objects. (Similar question for the other custom and template WhatsApp examples).

File: messages/whatsapp/send-button-link.php [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+                    ],
+                ],
+            ],
+            [
+                "type" => "body",
+                "parameters" => [
+                    ["type" => "text", "text" => "Anand"],
+                    ["type" => "text", "text" => "Quest"],
+                    ["type" => "text", "text" => "113-0921387"],
+                    ["type" => "text", "text" => "23rd Nov 2019"],
```
Comment (superchilled) [inline messages/whatsapp/send-button-link.php:34]: Is having a comma after the last element in an array a PHP convention?

**Review by superchilled — COMMENTED** [review]
Looks good overall. I've suggested some different variable names, and also added a couple of questions.

File: messages/whatsapp/send-button-link.php (hunk 21 above)
Comment (SecondeJK) [reply messages/whatsapp/send-button-link.php:34]: You can have either, but more modern version coding standards do encourage it to be added

File: messages/whatsapp/send-button-link.php (hunk 20 above)
Comment (SecondeJK) [reply messages/whatsapp/send-button-link.php:19]: The PHP array here is directly converted from the JSON example payload - so yeah what the client does in order to send the body is convert it to JSON

**Review by superchilled — COMMENTED** [review]
Missed a few of the var name changes.

**Review by superchilled — APPROVED** [review]

**Pull request merged**